src/mps.jl

```diff
@@ -655,6 +655,8 @@ function sample(rng::AbstractRNG, m::MPS)
     error("sample: MPS is not normalized, norm=$(norm(m[1]))")
   end
 
+  ElT = promote_itensor_eltype(m)
```

Comment:
I've been preferring `scalartype(m)` for this these days, probably we will remove `promote_itensor_eltype`.